mmcls/models/backbones/efficientnet.py

```diff
@@ -0,0 +1,300 @@
+              [[3, 48, 0, 0, 2, 8, 1],
+               [3, 48, 0, 0, 1, 8, 1],
+               [3, 48, 0, 0, 1, 8, 1],
+               [3, 48, 0, 0, 1, 8, 1]],
+              [[5, 96, 0, 0, 2, 8, 0],
+               [5, 96, 0, 0, 1, 8, 0],
+               [5, 96, 0, 0, 1, 8, 0],
+               [5, 96, 0, 0, 1, 8, 0],
+               [5, 96, 0, 0, 1, 8, 0],
+               [5, 144, 0, 0, 1, 8, 0],
+               [5, 144, 0, 0, 1, 8, 0],
+               [5, 144, 0, 0, 1, 8, 0],
+               [5, 144, 0, 0, 1, 8, 0]],
+              [[5, 192, 0, 0, 2, 8, 0],
+               [5, 192, 0, 0, 1, 8, 0]],
+              [[1, 1280, 0, 0, 1, 0, -1]]
+              ]
+    }   # yapf: disable
+
+    # Parameters to build different kinds of architecture.
+    # From left to right: scaling factor for width, scaling factor for depth,
+    # resolution, dropout ratio.
+    arch_settings = {
+        'b0': (1.0, 1.0, 224, 0.2),
+        'b1': (1.0, 1.1, 240, 0.2),
+        'b2': (1.1, 1.2, 260, 0.3),
+        'b3': (1.2, 1.4, 300, 0.3),
+        'b4': (1.4, 1.8, 380, 0.4),
+        'b5': (1.6, 2.2, 456, 0.4),
+        'b6': (1.8, 2.6, 528, 0.5),
+        'b7': (2.0, 3.1, 600, 0.5),
+        'b8': (2.2, 3.6, 672, 0.5),
+        'es': (1.0, 1.0, 224, 0.2),
+        'em': (1.0, 1.1, 240, 0.2),
+        'el': (1.2, 1.4, 300, 0.3)
+    }
+
+    def __init__(self,
+                 arch='b0',
+                 out_indices=(6, ),
+                 frozen_stages=0,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN'),
+                 act_cfg=dict(type='Swish'),
+                 norm_eval=False,
+                 with_cp=False):
+        super(EfficientNet, self).__init__()
+        assert arch in self.arch_settings
+        self.arch_setting = self.arch_settings[arch]
+        self.layer_setting = self.layer_settings[arch[:1]]
+        for index in out_indices:
+            if index not in range(0, len(self.layer_setting)):
+                raise ValueError('the item in out_indices must in '
+                                 f'range(0, {len(self.layer_setting)}). '
+                                 f'But received {index}')
+
+        if frozen_stages not in range(len(self.layer_setting) + 1):
+            raise ValueError('frozen_stages must be in range(0, '
+                             f'{len(self.layer_setting) + 1}). '
+                             f'But received {frozen_stages}')
+        self.out_indices = out_indices
+        self.frozen_stages = frozen_stages
+        self.conv_cfg = conv_cfg
+        self.norm_cfg = norm_cfg
+        self.act_cfg = act_cfg
+        self.norm_eval = norm_eval
+        self.with_cp = with_cp
+
+        self.layer_setting = model_scaling(self.layer_setting,
+                                           self.arch_setting)
+        self.dropout_rate = self.arch_setting[3]
+        self.in_channels = make_divisible(self.layer_setting[0][0][1], 8)
+        self.out_channels = self.layer_setting[-1][-1][1]
```

Comment:
Shall we use dict with meaningful keys for each element in `layer_setting`? `[0][0][1]` and `[-1][-1][1]` may not be that straightforward for users.

Reply:
I use ` block_cfg_0 = self.layer_setting[0][0]` and `block_cfg_last = self.layer_setting[-1][0]` to represent block settings of the first and the last layer. For the blocks in the middle blocks,  `(kernel_size, out_channels, se_ratio, use_swish, stride, expand_ratio, block_type) = block_cfg` has been used to give meanings to each element.
Can this way make it more clear?

Reply:
Thanks. It is more clear to me now.